pennylane/transforms/decompose.py

```diff
@@ -283,6 +287,17 @@ def eval(self, jaxpr: jax.extend.core.Jaxpr, consts: Sequence, *args) -> list:
                     subfuns, params = eq.primitive.get_bind_params(eq.params)
                     outvals = eq.primitive.bind(*subfuns, *invals, **params)
 
+                    if (
+                        self._num_available_work_wires is not None
+                        and eq.primitive.name == "allocate"
+                    ):
+                        self._num_available_work_wires -= params["num_wires"]
+                    if (
+                        self._num_available_work_wires is not None
+                        and eq.primitive.name == "deallocate"
+                    ):
+                        self._num_available_work_wires += len(invals)
```

Comment:
These two branches are exclusive to each but share the same entry `self._num_available_work_wires is not None`. For being more straightforward let's nest the two condition level of `self._num_available_work_wires` and `eq.primitive.name`
```suggestion
                    if self._num_available_work_wires is not None:
                        if eq.primitive.name == "allocate":
                            self._num_available_work_wires -= params["num_wires"]
                        if eq.primitive.name == "deallocate":
                            self._num_available_work_wires += len(invals)
```

Reply:
Alright I just noticed how deep this part has been already nested. I'm shocked by why this could not have been avoided. Anyway, feel free to avoid my readability suggestions since they won't add much to it even if accepted